rustls/src/tls12/mod.rs

```diff
@@ -169,6 +66,36 @@ impl fmt::Debug for Tls12CipherSuite {
     }
 }
 
+pub(crate) trait Tls12AeadAlgorithm: Send + Sync + 'static {
+    fn decrypter(&self, key: &[u8], iv: &[u8]) -> Box<dyn MessageDecrypter>;
+    fn encrypter(&self, key: &[u8], iv: &[u8], extra: &[u8]) -> Box<dyn MessageEncrypter>;
+    fn key_block_shape(&self) -> KeyBlockShape;
+    #[cfg(feature = "secret_extraction")]
+    fn extract_keys(&self, key: &[u8], iv: &[u8], explicit: &[u8]) -> ConnectionTrafficSecrets;
+}
+
+/// How a TLS1.2 `key_block` is partitioned.
+///
+/// nb. ciphersuites with non-zero `mac_key_length` not currently supported
+pub(crate) struct KeyBlockShape {
+    /// How long keys are.
+    ///
+    /// `enc_key_len` terminology is from the standard.
```

Comment:
nit: maybe replace "the standard" with RFC 5246 A.6? 

Also being highly pedantic the RFC terminology uses `length` throughout instead of `len`, e.g. `enc_key_length` and `fixed_iv_length`.